keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensor.java

```diff
@@ -482,7 +480,7 @@ public DoubleTensor powInPlace(double exponent) {
 
     @Override
     public DoubleTensor sqrtInPlace() {
-        return pow(0.5);
+        return powInPlace(0.5);
```

Comment:
yikes! how did this get past our tests?

Reply:
We actually don't have much in the way of tests for inPlace vs non in place (partially because a lot of the inPlace stuff doesn't work anyway and we have to return a new tensor).  Because of this (and because I want us to refactor the Tensor API anyway) - I didn't add any tests for this explicitly (and I don't think there's much point in doing so either).

The revised tests do get a lot more scalar coverage than we had before though.